plugins/blockupdates/main.go

```diff
@@ -0,0 +1,421 @@
+	}
+	result["destructs"] = destructs
+	accounts := make(map[string]hexutil.Bytes)
+	for k, v := range su.Accounts {
+		accounts[k.String()] = hexutil.Bytes(v)
+	}
+	result["accounts"] = accounts
+	storage := make(map[string]map[string]hexutil.Bytes)
+	for m, s := range su.Storage {
+		storage[m.String()] = make(map[string]hexutil.Bytes)
+		for k, v := range s {
+			storage[m.String()][k.String()] = hexutil.Bytes(v)
+		}
+	}
+	result["storage"] = storage
+	code := make(map[string]hexutil.Bytes)
+	for k, v := range su.Code {
+		code[k.String()] = hexutil.Bytes(v)
+	}
+	result["code"] = code
+	return json.Marshal(result)
+}
+
+// EncodeRLP converts the stateUpdate to a storedStateUpdate, and RLP encodes the result for storage
+func (su *stateUpdate) EncodeRLP(w io.Writer) error {
+	destructs := make([]common.Hash, 0, len(su.Destructs))
+	for k := range su.Destructs {
+		destructs = append(destructs, k)
+	}
+	accounts := make([]kvpair, 0, len(su.Accounts))
+	for k, v := range su.Accounts {
+		accounts = append(accounts, kvpair{k, v})
+	}
+	s := make([]storage, 0, len(su.Storage))
+	for a, m := range su.Storage {
+		accountStorage := storage{a, make([]kvpair, 0, len(m))}
+		for k, v := range m {
+			accountStorage.Data = append(accountStorage.Data, kvpair{k, v})
+		}
+		s = append(s, accountStorage)
+	}
+	code := make([]kvpair, 0, len(su.Code))
+	for k, v := range su.Code {
+		code = append(code, kvpair{k, v})
+	}
+	return rlp.Encode(w, storedStateUpdate{destructs, accounts, s, code})
+}
+
+// DecodeRLP takes a byte stream, decodes it to a storedStateUpdate, the n converts that into a stateUpdate object
+func (su *stateUpdate) DecodeRLP(s *rlp.Stream) error {
+	ssu := storedStateUpdate{}
+	if err := s.Decode(&ssu); err != nil { return err }
+	su.Destructs = make(map[common.Hash]struct{})
+	for _, s := range ssu.Destructs {
+		su.Destructs[s] = struct{}{}
+	}
+	su.Accounts = make(map[common.Hash][]byte)
+	for _, kv := range ssu.Accounts {
+		su.Accounts[kv.Key] = kv.Value
+	}
+	su.Storage = make(map[common.Hash]map[common.Hash][]byte)
+	for _, s := range ssu.Storage {
+		su.Storage[s.Account] = make(map[common.Hash][]byte)
+		for _, kv := range s.Data {
+			su.Storage[s.Account][kv.Key] = kv.Value
+		}
+	}
+	su.Code = make(map[common.Hash][]byte)
+	for _, kv := range ssu.Code {
+		su.Code[kv.Key] = kv.Value
+	}
+	return nil
+}
+
+type externalProducerBlockUpdates interface {
+	BlockUpdates(*gtypes.Block, *big.Int, gtypes.Receipts, map[common.Hash]struct{}, map[common.Hash][]byte, map[common.Hash]map[common.Hash][]byte, map[common.Hash][]byte)
+}
+
+type externalProducerPreReorg interface {
+	BUPreReorg(common.Hash, []common.Hash, []common.Hash)
+}
+
+type externalProducerPostReorg interface {
+	BUPostReorg(common.Hash, []common.Hash, []common.Hash)
+}
+
+type blockUpdatesModule struct {
+	backend types.Backend
+}
+
+type blockUpdatesAPI struct {
+	backend types.Backend
+}
+
+func (*blockUpdatesModule) ExternUpdatesTest() string {
```

Comment:
I'm still not sure I understand what we're achieving by having this. Let's discuss at EOD.